frontend/server/src/test/java/org/pytorch/serve/ModelServerTest.java

```diff
@@ -2088,8 +2089,38 @@ public void testUnregisterModelFailure() throws InterruptedException, SkipExcept
 
         channel = TestUtils.connect(ConnectorType.MANAGEMENT_CONNECTOR, configManager);
         Assert.assertNotNull(channel);
+        TestUtils.setResult(null);
+        TestUtils.setLatch(new CountDownLatch(1));
         TestUtils.unregisterModel(channel, "noopversioned", "1.11", false);
+        TestUtils.getLatch().await();
+
+        TestUtils.setResult(null);
+        TestUtils.setLatch(new CountDownLatch(1));
         TestUtils.unregisterModel(channel, "noopversioned", "1.2.1", false);
+        TestUtils.getLatch().await();
+    }
+
+    @Test(
+            alwaysRun = true,
+            dependsOnMethods = {"testUnregisterModelFailure"})
+    public void testClientTimeout() throws InterruptedException {
+        Channel mgmtChannel = TestUtils.connect(ConnectorType.MANAGEMENT_CONNECTOR, configManager);
+        loadTests(mgmtChannel, "echo-client-timeout.mar", "echo-client-timeout");
+
+        Channel inferChannel = TestUtils.connect(ConnectorType.INFERENCE_CONNECTOR, configManager);
+        TestUtils.setResult(null);
+        TestUtils.setLatch(new CountDownLatch(1));
+        DefaultFullHttpRequest req =
+                new DefaultFullHttpRequest(
+                        HttpVersion.HTTP_1_1, HttpMethod.POST, "/predictions/echo-client-timeout");
+        req.content().writeZero(10385760);
+        HttpUtil.setContentLength(req, req.content().readableBytes());
+        req.headers().set(HttpHeaderNames.CONTENT_TYPE, HttpHeaderValues.APPLICATION_OCTET_STREAM);
+        inferChannel.writeAndFlush(req);
+        TestUtils.getLatch().await(1, TimeUnit.SECONDS);
+        Assert.assertNull(TestUtils.result);
+
```

Comment:
what's the expected returned error code?

Reply:
nothing return if client already timeout.